lib/cloudant.js

```diff
@@ -742,23 +737,33 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   debug('Cloudant.prototype.updateOrCreate %j %j', model, data);
   var self = this;
   var idName = self.idName(model);
-  var id = data[idName];
+  var mo = self.selectModel(model);
+  var id = data[idName].toString();
+
+  // Callback handler for both create calls.
+  var createHandler = function(err, id) {
+    if (err) return cb(err);
+    mo.db.get(id, function(err, doc) {
+      if (err) return cb(err);
+      return cb(err, self.fromDB(model, mo, doc), {isNewInstance: true});
+    });
+  };
+
   if (id) {
     self.updateAttributes(model, id, data, {}, function(err, docs) {
-      if (err && err.statusCode && err.statusCode === 404) {
-        self.create(model, data, {}, function(err) {
-          if (err) return cb(err);
-          return cb(err, data, {isNewInstance: true});
-        });
+      if (err && err.statusCode !== 404) return cb(err);
+      else if (err && err.statusCode && err.statusCode === 404) {
```

Comment:
If `err.statusCode` didn't exist, then the if statement would've been true and it would've returned.
Checking `err.statusCode` should be fine at this point.